api/Catalog.go

```diff
@@ -0,0 +1,89 @@
+package api
+
+import (
+	"encoding/json"
+	"github.com/aditya/ProjectCatalog/models"
+	"github.com/aditya/ProjectCatalog/services"
+	"github.com/gorilla/mux"
+	"net/http"
+)
+
+type CatalogController struct{
+	CatalogService services.ICatalogService
+}
+
+func (h *CatalogController) CreateProduct(w http.ResponseWriter, r *http.Request){
+	w.Header().Set("Content-Type", "application/json")
+	var newProduct models.Product
+	_ = json.NewDecoder(r.Body).Decode(&newProduct)
+	err := h.CatalogService.CreateProduct(newProduct)
+	if err==nil{
+		json.NewEncoder(w).Encode("Product added successfully")
+		return
+	}
+
+	json.NewEncoder(w).Encode("Encountered error while adding product, product already exists")
+}
+
+func (h *CatalogController) ShowProduct(w http.ResponseWriter,r *http.Request){
+	w.Header().Set("Content-Type", "application/json")
+	product:=h.CatalogService.ShowProduct()
+	json.NewEncoder(w).Encode(product)
+}
+
+func (h *CatalogController) ShowProductById(w http.ResponseWriter, r *http.Request){
+	w.Header().Set("Content-Type", "application/json")
+	params:=mux.Vars(r)
+	product:=h.CatalogService.ShowProductById(params["id"])
+	emptyProduct:=models.Product{}
+	if product==emptyProduct{
+		json.NewEncoder(w).Encode("Item does not exists")
+	}
+	json.NewEncoder(w).Encode(product)
+}
+
+func (h *CatalogController) UpdateProduct(w http.ResponseWriter, r *http.Request){
+	w.Header().Set("Content-Type", "application/json")
+	params:=mux.Vars(r)
+	var updatedProduct models.Product
+	_ = json.NewDecoder(r.Body).Decode(&updatedProduct)
```

Comment:
if there is an error while decoding the request body, we should return HTTP code 400 (bad request).